### ISSUE TITLE
SRT streams have trouble connecting/reconnecting while Recording is running

### ISSUE THREAD

**Issue by loopy750** [obsproject/obs-studio#4596]
### Operating System Info

Windows 10

### OBS Studio Version

27.0.0-rc2

### OBS Studio Log URL

https://obsproject.com/logs/YvNqZDkccJqVfIK-

### Expected Behavior

Streams should be able to reconnect without issue.

### Current Behavior

SRT stream from Larix on Android fails to reconnect when stopped while OBS is recording.

### Steps to Reproduce

1. Open OBS with Media Source containing SRT URL.
2. Launch external stream app, and start recording in OBS
3. Restart stream. Won't reconnect.
4. Stop recording in OBS. Stream will instantly reconnect.

### Anything else we should know?

Interestingly with 26.1.1 it has a hard time reconnecting regardless of recording status. So SRT stability has improved in v27, but still not 100%. (Opinion: as SRT is all about providing stability and a robust stream under any circumstances, hopefully this and any other stability related issues can be resolved, else users might not trust being able to leave OBS/SRT unattended.)

**Comment by pkviet** [comment]
Try to add this to your Media Source URL in obs:
?mode=listener&listener_timeout=5000000&timeout=5000000

**Comment by loopy750** [comment]
> 
> 
> Try to add this to your Media Source URL in obs:
> ?mode=listener&listener_timeout=5000000&timeout=5000000

Thanks for the suggestion but unfortunately it doesn't help. Also I think you meant `listen_timeout`, but either way, no go.

Although, I can say that `timeout=5000000` may have helped somewhat, as there are times it will reconnect, but more often than not, I can have it recording for several minutes while it's trying to reconnect, but only after I end the recording, the stream is able to then reconnect.

So even though it's doing something, it's definitely not a 100% fix for this issue.

**Comment by kpcenti** [comment]
I have the same problem even if I'm recording or not.

**Comment by pkviet** [comment]
Did you try what I suggested above @kpcenti  ?

**Comment by kpcenti** [comment]
Yes I have, it's better but still has the problems from time to time. I also tried it on the newest RC version and it's the same

**Comment by HyperVolcan** [comment]
This issue still exists SRT stream cannot reconnect if connection is lost while OBS is recording

**Comment by timtam85465414** [comment]
Hello, this issue still exist...will it be fixed one day...? any devs?

**Comment by RytoEX** [comment]
I can confirm this issue on OBS Studio 27.1.3 on Windows.  The Steps to Reproduce must be performed in exactly the order specified, and the active output must be recording.  This does not seem to occur with an active stream or virtual camera.

**Comment by DiegoTS55** [comment]
Hello,

 same reconnecting issue.

Here my config:

OBS Studio 27.1.3 (64 bit) running on macos 11.4 ( Intel processor)
Larix Broadcaster Version 1.1.18 build 251  running on iPhone 13,  ios 15.1.1


OBS Settings:

Media Source input
srt://XXX.XXX.XXX.XXX:XXXXX?mode=listener
Network Buffering 2 MB
Connect Delay   1Sec
all other options de-flagged

Issue:

 if transmission is interrupted the displayed image on OBS freezes. After 60secs it restarts automatically ( it looks like a default timeout somewhere in OBS)

Tested with different srt parameters as the  "timeout" one. Sometime works better but it is not consistent and I do not understand which parameter/parameters changes the relation.

It looks like  an OBS related issue.

Any suggestion is very appreciated.

Thanks.

**Comment by DiegoTS55** [comment]
> Hello,
> 
> same reconnecting issue.
> 
> Here my config:
> 
> OBS Studio 27.1.3 (64 bit) running on macos 11.4 ( Intel processor) Larix Broadcaster Version 1.1.18 build 251 running on iPhone 13, ios 15.1.1
> 
> OBS Settings:
> 
> Media Source input srt://XXX.XXX.XXX.XXX:XXXXX?mode=listener Network Buffering 2 MB Connect Delay 1Sec all other options de-flagged
> 
> NO recording NO streaming
> 
> Issue:
> 
> if transmission is interrupted the displayed image on OBS freezes. After exactly 60secs it restarts automatically. During these 60 secs Larix Broadcaster keeps the transmission alive sending data, on OBS the image remains freezed; at the end of the 60 secs it seems the connection/trasmission is interrupted and Larix Broadcaster reconnects again to OBS and then srt is displayed correctly.
> 
> Tested with different srt parameters( as the "timeout" ). Sometime works better but it is not consistent and I do not understand which parameter/parameters changes the relation.
> 
> It looks like an OBS related issue.
> 
> Any suggestion is very appreciated.
> 
> Thanks.
Well, continuing testing.

Here some other observations

in OBS sorce input "media source" settings:

a) Media Source input
   srt://XXX.XXX.XXX.XXX:XXXXX?mode=listener
   Network Buffering 2 MB
   Connect Delay 1 Sec
   all other options de-flagged

If SRT stream is interrupted the displayed image on OBS freezes. Restart Automatically after 60 secs

b) Media Source input
   srt://XXX.XXX.XXX.XXX:XXXXX?mode=listener
   Network Buffering 2 MB
   Connect Delay 15 Sec
   all other options de-flagged

If SRT stream is interrupted the displayed image on OBS freezes. Restart Automatically after 75 secs

c) Media Source input
   srt://XXX.XXX.XXX.XXX:XXXXX?mode=listener
   Network Buffering 2 MB
   Connect Delay 30 Sec
   all other options de-flagged

If SRT stream is interrupted the displayed image on OBS freezes. Restart Automatically after 90 secs..
It looks like that the Connect Delay Slider default time of 1 sec is wrong because it correspond to 60 secs.

Please verify on your configuration if you can reproduce.

Thanks

**Comment by kamileq** [comment]
I have the same issue on OBS 27.1.3 on macOS like DiegoTS55. Using SRT Listener is unreliable in such condition. I found solution:
a) using SLS localhost server on macOS
b) using OBS in Windows 10 - much more stable option
c) using OBS in Windows 10 and SRT Mini Server - very stable option but then canNOT use Loopy SRT Stats Monitor

**Comment by pkviet** [comment]
i've repro'd with OBS 27.2 beta 4. I'll investigate.

**Comment by pkviet** [comment]
it seems I get the issue too when I stream.

**Comment by robotmay** [comment]
Can confirm this myself and that Larix broadcaster isn't the issue - I have an SRT relay in between the two and the logs reflect that OBS takes roughly 60 seconds to reconnect once the stream drops.

**Comment by robotmay** [comment]
Ah ha! It's the network buffering setting. If you set it to 0MB it drops and reconnects almost immediately.

**Comment by Lampburner** [comment]
This seems to still be an issue in OBS 27.2.3. SRT streams from Larix Broadcaster will not reconnect in OBS while I am recording locally.

**Comment by RytoEX** [comment]
Does this still occur in OBS Studio 28.1.2 (or newer)?

**Comment by Lampburner** [comment]
Yes, I'm still experiencing the same issue in OBS Studio 28.1.2 when I
disconnect the video feed from Larix Broadcaster and try to reconnect while
recording in OBS. Failure to reconnect.

**Comment by jackhilo** [comment]
I have attempted to use the solutions in this thread and will say it is still a present issue; I am currently unable to re-connect when recording and have not been able to successfully reconnect once in OBS 29.0.0 it's very sad to see this is just going untouched as Local OBS recordings are my primary way to save videos to edit later.

**Comment by zdnguyen** [comment]
Also on OBS 29.0 having the same issues. SRT from LAN will occasionally reconnect while recording from Larix/IRL pro, but on a different network it has 0 success trying to connect or reconnect while recording.

**Comment by alexandre-leites** [comment]
I also have this issue. Reconnection do not work until I stop recording.

**Comment by pkviet** [comment]
here's a workaround:
- set Media Source to be a caller
- set Larix to be a listener
- reconnect then works even when recording.
I investigated a bit and the issue the other way around seems related to a bug in FFmpeg where the srt socket is not released when there's been a disconnection.
How the hell having a recording achieves that, I have no idea. It's very very weird.

**Comment by loopy750** [comment]
Coming back to this, I wasn't able to reproduce it, then I realised why.

> set Media Source to be a caller
set Larix to be a listener

I couldn't get it working by reversing the `caller` and `listener`, but I worked out what my unintentional workaround was:

Create a scene, call it "Dummy SRT" or whatever, and add a Media Source with a port you won't use, eg: `srt://127.0.0.1:60000?mode=listener`. All boxes are unchecked but I don't know if that makes a difference.

Now Recording has no effect.

**Comment by pkviet** [comment]
@_@ weird. I'd rather have this bug solved. I wonder if an srt update might help. Anyway, if you can live with this workaround for the time being, that's good.

**Comment by loopy750** [comment]
I completely agree, I'd rather see the bug sorted. That being said, if my (or your) workaround works for others, then it's as good as a bug fix, at only takes a few seconds to apply.

**Comment by Alivea34** [comment]
Ok, I was pulling my hair out trying to figure this out. I got it working on my end. First i don't know if it matters but I added ?mode=listener&listen_timeout=5000000&timeout=5000000 to the OBS side. The magic bullet for me was finding a UDP port. I changed my port to 2000 and wallah fixed. tried it with CameraFi Live, Larix Broadcaster, Haivision Play Pro. it works on all of them. Caller on camera side and listener on OBS side. Hope it helps you.

**Comment by Alivea34** [comment]
Nevermind. I found what I did that made a loop hole. I started the broadcast on port 8000, hit record, stopped the broadcast, changed the port in Broadcaster, changed the port in OBS, then reconnected all while it was still recording. THEN I was able to start and stop broadcast all I wanted. it seems the port you first hit record on is somehow locked while recording while other ports can start and stop no problem. UGHHH I just need a reliable solution. how has this not been figured out yet?

**Comment by richtopia** [comment]
This issue persists in 30.2.2. I'm commenting here for anyone else searching for a work around: I run a Mist Server (https://mistserver.org) on my PC with OBS to ingest the SRT stream and rebroadcast as an MP4 html5 video. A bit convoluted and induces some latency, but it does handle reconnects without issue.

**Comment by adsa95** [comment]
OBS 31.0.1, issue still persists. Makes SRT useless for streaming from a mobile device as it requires very stable connection.
